pkg/controllers/inventory/inventory_controller.go

```diff
@@ -0,0 +1,216 @@
+package inventory
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/config"
+	"github.com/vmware-tanzu/nsx-operator/pkg/logger"
+	commonservice "github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/common"
+	"github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/inventory"
+)
+
+const (
+
+	// How long to wait before retrying
+	minRetryDelay = 5 * time.Second
+	maxRetryDelay = 300 * time.Second
+
+	inventoryGCJitterFactor = 0.1
+)
+
+var (
+	log     = &logger.Log
+	keyFunc = cache.DeletionHandlingMetaNamespaceKeyFunc
+)
+
+type InventoryController struct {
+	Client               client.Client
+	service              *inventory.InventoryService
+	inventoryObjectQueue workqueue.TypedRateLimitingInterface[any]
```

Comment:
We can define a thread-safe KeySet using sync.Map, avoiding putting inventoryMutex everywhere.

Reply:
inventoryMutex is not only for KeySet. it's used to guarantee that only one worker handle the inventory